src/wwwroot/app/services/LogginService.js

```diff
@@ -56,97 +60,122 @@ function isArrayCase(log) {
+      });
+      let pastTitles = map(pastValues, thesaurusId => {
          return find(thesaurus, ['id', parseInt(thesaurusId)]).title;
       });
       return {
          field: /city/i.test(field) ? 'Cities' : 'Levels',
-         value: titles.length ? titles.join(', ') : EMPTY
+         newValue: newTitles.length ? newTitles.join(', ') : EMPTY,
+         pastValue: pastTitles.length ? pastTitles.join(', ') : EMPTY
       };
    });
 }
 function fromSimple(log) {
    if (/id/i.test(log.field)) {
       log.field = trim(log.field, 'Id');
    }
-   return getNewValue(log).then((newValueOfField) => {
+   return getNewAndPastValue(log).then((valuesContainer) => {
       return {
          createdOn: log.createdOn,
-         user: log.user.lastName,
+         user: `${log.user.lastName} ${log.user.firstName}`,
          field: log.field,
-         value: newValueOfField
+         newValue: valuesContainer.newValue,
+         pastValue: valuesContainer.pastValue
       };
    });
 }
 
-function getNewValue(log) {
-   let deffered = _$q.defer();
+function getThesaurusTitle(thesaurusName, id) {
+   return _ThesaurusService.getThesaurusTopic(thesaurusName, id).then(thesaurus => {
+      return thesaurus.title;
+   });
+}
+function getNewAndPastForThesaurus(thesaurusName, log) {
+   return _$q.all([
+      getThesaurusTitle(thesaurusName, parseInt(head(log.newValues))),
+      getThesaurusTitle(thesaurusName, parseInt(head(log.pastValues)))
+   ]).then(valuesContainer => {
+      return {
+         newValue: valuesContainer[VALUES_INDEXES.NEW_VALUE],
+         pastValue: valuesContainer[VALUES_INDEXES.PAST_VALUE]
+      };
+   });
+}
+
+function getNewAndPastValue(log) {
    if (/city/i.test(log.field)) {
-      let cityId = parseInt(head(log.values));
-      _ThesaurusService.getThesaurusTopic('city', cityId).then(city => {
-         deffered.resolve(city.title);
-      });
+      return getNewAndPastForThesaurus('city', log);
    } else if (/department/i.test(log.field)) {
-      let departmentId = parseInt(head(log.values));
-      _ThesaurusService.getThesaurusTopic('department', departmentId).then(department => {
-         deffered.resolve(department.title);
-      });
+      return getNewAndPastForThesaurus('department', log);
    } else if (/responsible/i.test(log.field)) {
-      let responsibleId = parseInt(head(log.values));
-      _UserService.getUserById(responsibleId).then(user => {
-         deffered.resolve(user.lastName);
+      let newResponsibleId = parseInt(head(log.values));
+      return _UserService.getUserById(newResponsibleId).then(user => {
+         return user.lastName;
       });
    } else {
+      debugger;
       if (/date/i.test(log.field)) {
-         log.values = map(log.values, val => {
+         log.newValues = map(log.newValues, val => {
+            if (moment(val).isValid()) {
+               return moment(val).toISOString();
+            } else {
+               return val;
+            }
+         });
```

Comment:
I think you can change it to something like that: 
return moment(val).isValid() ? moment(val).toISOString() : val;